src/lib/plugin.js

```diff
@@ -25,77 +26,102 @@ const assertOptionType = (opts, field, type) => {
 }
 
 module.exports = class PluginVirtual extends EventEmitter2 {
-
-  constructor (opts) {
+  constructor (PaymentChannelBackend, opts) {
     super()
 
-    assertOptionType(opts, 'currencyCode', 'string')
-    assertOptionType(opts, 'currencyScale', 'number')
-    assertOptionType(opts, 'maxBalance', 'string')
-    assertOptionType(opts, 'secret', 'string')
-    assertOptionType(opts, 'peerPublicKey', 'string')
-    assertOptionType(opts, '_store', 'object')
-    assertOptionType(opts, 'rpcUri', 'string')
-
-    this._secret = opts.secret
-    this._peerPublicKey = opts.peerPublicKey
-    this._publicKey = Token.publicKey(this._secret)
-    this._currencyCode = opts.currencyCode.toUpperCase()
-    this._currencyScale = opts.currencyScale
+    this._opts = opts
+    this._stateful = !!(opts._backend || opts._store)
 
-    this._store = opts._store
-    this._maxBalance = opts.maxBalance
-    this._balance = new Balance({
-      maximum: this._maxBalance,
-      store: this._store
-    })
+    if (this._stateful) {
+      assertOptionType(opts, 'currencyCode', 'string')
+      assertOptionType(opts, 'currencyScale', 'number')
+      assertOptionType(opts, 'maxBalance', 'string')
```

Comment:
You mean it defaults to the string '-Infinity', no? If `minBalance` is optional, I would make its input validation conditional. For example:

```javascript
if (opts.minBalance) {
  assertOptionType(opts, 'minBalance', 'string')
}
```